### ISSUE TITLE
Configurable order of datastreams in Visualizer

### ISSUE THREAD

**Issue by rgolovanov** [rerun-io/rerun#6010]
Currently, visualized data has fixed order in the Visualizer. For example, when draw multiple curves on same 2d plot it is impossible to change the order - this makes the analysis of data difficult and requires toggle visibility of overlapped curves.

See https://www.rerun.io/viewer?url=https%3A%2F%2Fapp.rerun.io%2Fversion%2F0.15.1%2Fexamples%2Fplots.rrd
It is impossible to change order of `sin` and `cos`:
![image](https://github.com/rerun-io/rerun/assets/14174584/3a7c888d-3c47-4e5a-a4b3-0586933113c7)

**Comment by Wumpf** [comment]
This bottoms out to supporting the `DrawOrder` component for time series elements 👍

**Comment by rgolovanov** [comment]
@Wumpf it is more about dynamic draw order control via UI as we may not know which order is better at the code level.
I think same proposal shall be considered for 2D space view:
https://www.rerun.io/viewer?url=https%3A%2F%2Fapp.rerun.io%2Fversion%2F0.15.1%2Fexamples%2Fhuman_pose_tracking.rrd
![image](https://github.com/rerun-io/rerun/assets/14174584/2f1d31a4-dffc-4ba6-9b04-cb3673cb7ad6)

**Comment by Wumpf** [comment]
Makes sense. I also think that draw order component (like any other component!) should be a thing one can change from the ui! But good point that this is the main important thing here and likely needs special consideration on how that's shown 👍
